api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -205,30 +240,43 @@ message ScopedRoutesConfigDump {
     string version_info = 2;
 
     // The scoped route configurations.
-    repeated api.v2.ScopedRouteConfiguration scoped_route_configs = 3;
+    repeated api.v2.ScopedRouteConfiguration scoped_routes = 3;
 
-    // The timestamp when the scoped route config set was last updated.
+    // The timestamp when the scoped route config set was last successfully updated.
     google.protobuf.Timestamp last_updated = 4;
   }
 
+  message DynamicScopedRoute {
+    // The name or unique id of this route, pulled from config.
+    string name = 1;
+
+    // The scoped route state for any active route by this name.
+    DynamicScopedRouteState scoped_route_state = 2;
+
+    google.protobuf.Timestamp last_updated = 3;
+  }
+
   // The statically loaded scoped route configs.
-  repeated InlineScopedRouteConfigs inline_scoped_route_configs = 1;
+  repeated InlineScopedRoutes inline_scoped_routes = 1;
 
   // The dynamically loaded scoped route configs.
-  repeated DynamicScopedRouteConfigs dynamic_scoped_route_configs = 2;
+  repeated DynamicScopedRoute dynamic_scoped_routes = 2;
+
+  // Set if the last update failed, cleared after the next successful update.
+  UpdateFailureState error_state = 3;
```

Comment:
@stevenzzzz should this be on the `DynamicScopedRoute` granularity? I think we decided this API was basically a CDS-style state-of-the-world, so each individual named entity might have a distinct reason for failed update.

Reply:
The per-route-configuration update error status is kept by the above DynamicRoute. For SRDS I think this is fine. There is only one error status for a SRDS delta/SotW update, for the SotW form, it's the first whatever error of that update, for the delta-form, it's currently a concatenated error string for all the parsing/validating errors in that update.

Reply:
Well, the same could be said about CDS. But, if you look at what is done there, we break apart the per-resource errors and report them individually. We should do the same for SRDS I think.

Reply:
I agree we can keep a error status in the per-route-configuration as well.
Currently the error status could be one of "validation/invalid/duplicate" etc, as the scopedRouteConfiguration is simply a pointer to RDS + a key-builder. (it's not that heavy weight as a Cluster).

Also noteworthy to populate per-RouteConfigurationConfig error status, in the SotW for CDS and SRDS alike, would be interesting as they return on the first error.